pkg/registry/core/service/storage/alloc.go

```diff
@@ -409,7 +425,20 @@ func (al *Allocators) allocIPs(service *api.Service, toAlloc map[api.IPFamily]st
 			if parsedIP == nil {
 				return allocated, errors.NewInternalError(fmt.Errorf("failed to parse service IP %q", ip))
 			}
-			if err := allocator.Allocate(parsedIP); err != nil {
+			var err error
+			if utilfeature.DefaultFeatureGate.Enabled(features.MultiCIDRServiceAllocator) {
+				// if dryRun is being used the allocator is implemented by ipallocator.dryRunRange
```

Comment:
we have an open discussion to `dryrun` implementation in the new allocator. I think this proves the point that we need to do it anyway, since eventually we want to remove the old allocator (and thus we will have to implement the dry run anyway).